pkg/event/controller/controller.go

```diff
@@ -119,23 +119,25 @@ func New(config *Config) (*Controller, error) {
 	return &ctl, nil
 }
 
-// Start will start the controller, and wait until stopCh receives a message
+// Start starts the controller.
 func (c *Controller) Start(stopCh <-chan struct{}) error {
-	klog.Info("Starting Controller")
+	klog.Info("Starting the Event controller...")
 
 	err := c.resourceWatcher.Start(stopCh)
 	if err != nil {
 		return err
 	}
 
-	klog.Info("Event Controller workers started")
-	<-stopCh
-	klog.Info("Event Controller stopping")
+	klog.Info("Event controller started")
+
```

Comment:
```suggestion
	klog.Info("Event controller started")
        go func() {
            <- stopCh
            c.Stop()
        }
```

Reply:
This is still problematic for the UTs, ie frequent race failures. The tests need to wait for `Stop` to complete. The simplest solution is to explicitly call `Stop` otherwise we need a method and flag or chan to check for stopped or wait for it. We can do the latter but seems like overkill. Also, in the production  code, calling `Stop` explicitly makes it deterministic that all handlers will be stopped.